tests/tekton-tasks_test.go

```diff
@@ -33,54 +35,106 @@ var _ = Describe("Tekton-tasks", func() {
 
 		It("[test_id:TODO]operator should not create any cluster tasks", func() {
 			liveTasks := &pipeline.ClusterTaskList{}
-			Eventually(func() bool {
-				err := apiClient.List(ctx, liveTasks,
-					client.MatchingLabels{
-						tektontasks.TektonTasksVersionLabel: operands.TektonTasksVersion,
-					},
-				)
-				Expect(err).ToNot(HaveOccurred())
-				return len(liveTasks.Items) == 0
-			}, tenSecondTimeout, time.Second).Should(BeTrue(), "tasks should not exists")
+
+			err := apiClient.List(ctx, liveTasks,
+				client.MatchingLabels{
+					common.AppKubernetesManagedByLabel: common.AppKubernetesManagedByValue,
+				},
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			//Nothing to check, end the test successfully
+			if len(liveTasks.Items) == 0 {
+				return
+			}
+
+			err = apiClient.Delete(ctx, &liveTasks.Items[0])
+			Expect(err).ToNot(HaveOccurred())
+
+			deletedTask := &pipeline.ClusterTask{}
+			Eventually(func() metav1.StatusReason {
+				err := apiClient.Get(ctx, client.ObjectKeyFromObject(&liveTasks.Items[0]), deletedTask)
+				Expect(err).To(HaveOccurred())
+				return errors.ReasonForError(err)
+			}, tenSecondTimeout, time.Second).Should(Equal(metav1.StatusReasonNotFound), "task should not exists")
 		})
 
 		It("[test_id:TODO]operator should not create any service accounts", func() {
 			liveSA := &v1.ServiceAccountList{}
-			Eventually(func() bool {
-				err := apiClient.List(ctx, liveSA,
-					client.MatchingLabels{
-						tektontasks.TektonTasksVersionLabel: operands.TektonTasksVersion,
-					},
-				)
-				Expect(err).ToNot(HaveOccurred())
-				return len(liveSA.Items) == 0
-			}, tenSecondTimeout, time.Second).Should(BeTrue(), "service accounts should not exists")
+
+			err := apiClient.List(ctx, liveSA,
+				client.MatchingLabels{
+					common.AppKubernetesManagedByLabel: common.AppKubernetesManagedByValue,
+				},
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			//Nothing to check, end the test successfully
+			if len(liveSA.Items) == 0 {
+				return
+			}
+
+			err = apiClient.Delete(ctx, &liveSA.Items[0])
+			Expect(err).ToNot(HaveOccurred())
+
+			deletedSA := &v1.ServiceAccount{}
+			Eventually(func() metav1.StatusReason {
```

Comment:
Wouldn't `Consistently` be more appropriate for this test to verify that the operator does not create new objects anymore?

https://pkg.go.dev/github.com/onsi/gomega#Consistently